File: docs/snippets/gpu/dynamic_shape.cpp
#include <openvino/runtime/core.hpp>

int main() {
//! [dynamic_shape]

// Read model
ov::Core core;
auto model = core.read_model("model.xml");

model->reshape({{ov::Dimension(-1), ov::Dimension(-1)}});

// compile model and create infer request
auto compiled_model = core.compile_model(model, "GPU");
auto infer_request = compiled_model.create_infer_request();
auto input = model->get_parameters().at(0);

// ...

// create input tensor with specific shape
ov::Tensor input_tensor(input->get_element_type(), {2, 177});

// ...

infer_request.set_tensor(input, input_tensor);
infer_request.infer();

//! [dynamic_shape]

return 0;
}

File: docs/snippets/gpu/dynamic_shape.py
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0


import openvino.runtime as ov

#! [dynamic_shape]
core = ov.Core()

model = core.read_model("model.xml")
model.reshape([-1, -1])

# compile model and create infer request
compiled_model = core.compile_model(model, "GPU")
infer_request = compiled_model.create_infer_request()

# create input tensor with specific shape
input_tensor = ov.Tensor(model.input().element_type, [2, 224])

# ...

infer_request.infer([input_tensor])

#! [dynamic_shape]
